apps/examples/iperf2/compat/delay.c

```diff
@@ -0,0 +1,326 @@
+#ifdef HAVE_NANOSLEEP
+    delay_nanosleep(usec);
+#else
+    delay_busyloop(usec);
+#endif
+#endif
+}
+
+#ifdef HAVE_NANOSLEEP
+// Can use the nanosleep syscall suspending the thread
+void delay_nanosleep (unsigned long usec) {
+    struct timespec requested, remaining;
+    requested.tv_sec  = 0;
+    requested.tv_nsec = usec * 1000L;
+    // Note, signals will cause the nanosleep
+    // to return early.  That's fine.
+    nanosleep(&requested, &remaining);
+}
+#endif
+
+#if defined (HAVE_NANOSLEEP) || defined (HAVE_CLOCK_GETTIME)
+static void timespec_add_ulong (struct timespec *tv0, unsigned long value) {
+    tv0->tv_nsec += value;
+    if (tv0->tv_nsec >= BILLION) {
+	tv0->tv_sec++;
+	tv0->tv_nsec -= BILLION;
+    }
+}
+#endif
+
+#ifdef HAVE_KALMAN
+// Kalman versions attempt to support delay request
+// accuracy over a minimum guaranteed delay by
+// prediciting the delay error. This is
+// the basic recursive algorithm.
+static void kalman_update (kalman_state *state, double measurement) {
+    //prediction update
+    state->p = state->p + state->q;
+    //measurement update
+    state->k = state->p / (state->p + state->r);
+    state->x = state->x + (state->k * (measurement - state->x));
+    state->p = (1 - state->k) * state->p;
+}
+#endif
+
+#ifdef HAVE_CLOCK_GETTIME
+// Delay calls for systems with clock_gettime
+// Working units are nanoseconds and structures are timespec
+static void timespec_add_double (struct timespec *tv0, double value) {
+    tv0->tv_nsec += (unsigned long) value;
+    if (tv0->tv_nsec >= BILLION) {
+	tv0->tv_sec++;
+	tv0->tv_nsec -= BILLION;
+    }
+}
+// tv1 assumed greater than tv0
+static double timespec_diff (struct timespec tv1, struct timespec tv0) {
+    double result;
+    if (tv1.tv_nsec < tv0.tv_nsec) {
+	tv1.tv_nsec += BILLION;
+	tv1.tv_sec--;
+    }
+    result = (double) (((tv1.tv_sec - tv0.tv_sec) * BILLION) + (tv1.tv_nsec - tv0.tv_nsec));
+    return result;
+}
+static void timespec_add( struct timespec *tv0, struct timespec *tv1)
+{
+    tv0->tv_sec += tv1->tv_sec;
+    tv0->tv_nsec += tv1->tv_nsec;
+    if ( tv0->tv_nsec >= BILLION ) {
+	tv0->tv_nsec -= BILLION;
+	tv0->tv_sec++;
+    }
+}
+static inline
+int timespec_greaterthan(struct timespec tv1, struct timespec tv0) {
+    if (tv1.tv_sec > tv0.tv_sec ||					\
+	((tv0.tv_sec == tv1.tv_sec) && (tv1.tv_nsec > tv0.tv_nsec))) {
+	return 1;
+    } else {
+	return 0;
+    }
+}
+// A cpu busy loop for systems with clock_gettime
+void delay_busyloop (unsigned long usec) {
+    struct timespec t1, t2;
+    clock_gettime(CLOCK_REALTIME, &t1);
+    timespec_add_ulong(&t1, (usec * 1000L));
+    while (1) {
+	clock_gettime(CLOCK_REALTIME, &t2);
+	if (timespec_greaterthan(t2, t1))
+	    break;
+    }
+}
+// Kalman routines for systems with clock_gettime
+#ifdef HAVE_KALMAN
+// Request units is microseconds
+// Adjust units is nanoseconds
+void delay_kalman (unsigned long usec) {
+    struct timespec t1, t2, finishtime, requested={0,0}, remaining;
```

Comment:
space after coma & '='

requested = {0, 0}

Reply:
it's opensource so It would be better not to apply coding rule